.github/workflows/check_on_push.yaml

```diff
@@ -0,0 +1,24 @@
+name: Run static analysis
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  run-static-analysis:
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request' && github.event.pull_request.head.repo.owner.login != 'tarantool'
```

Comment:
It seems, we can do it without hardcoding 'tarantool' organization name. Please, consider this [code](https://github.com/tarantool/smtp/blob/556393ac5a8906a47910331e70db17481f60326a/.github/workflows/testing.yml#L12-L17) and the link inside it.

Reply:
This condition can be useful when we decide to use our own workers.
Anyway removed it now.

Reply:
I didn't vote for removal, but for changing of `github.event.pull_request.head.repo.owner.login != 'tarantool'` to `github.event.pull_request.head.repo.full_name != github.repository`.

Reply:
The positive aspect is that it should work as expected for a pull request to a non-mainline repository (from one fork to another fork).

Reply:
This repository has only 3 forks and all of them are made by Tarantool's people. Why do you care about forks?

Reply:
That's the short recipe, which should work for all cases. Such recipes often walk from a repository to repository, so it would be nice to use a good default everywhere (if there are no specific requirements).

Reply:
Seems resolved.